### PR TITLE
Changes on hov routing behaviour

### PR DESCRIPTION
**Pull request by chaupow**
# Issue

fixes https://github.com/valhalla/valhalla/issues/1297

## Tasklist

 - [x] add tests
 - [x] review - you must request approval to merge any PR to master
 - [x] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [x] Add to release notes in valhalla-docs
 - [x] update relevant documentation if this is an API impacting change
 - [x] Run RAD tests to confirm that this does not break expected behavior (i.e. access)

### CODE REVIEW

**Comment by dnesbitt61** [comment]
Can/should the same logic be done for motorcycle?

**Comment by chaupow** [comment]
@dgearhart tagged you for review 😺 

I suggest going through the individual commits is easiest because the removal of trailing whitespaces makes the file changes look confusing.

#### open questions still:
- [ ] motorcycle
- [ ] how to add a test

#### not in scope of this PR but needs to be ticketed:
- [ ] `hov:forward` and `hov:backward` parsing

**Comment by dgearhart** [comment]
@chaupow i think you meant you asked @gknisely for review

File: lua/graph.lua [hunk 1]
```diff
@@ -1424,9 +1424,11 @@ function filter_tags_generic(kv)
 
     local only_hov_allowed = kv["hov"] == "designated"
     if only_hov_allowed then
-      for lane in (kv["hov:lanes"] .. '|'):gmatch("([^|]*)|") do
-        if lane and lane ~= "designated" then
-          only_hov_allowed = false
+      if kv["hov:lanes"] then
```
Comment (chaupow) [inline lua/graph.lua]: goot catch

File: lua/graph.lua [hunk 2]
```diff
@@ -1417,11 +1417,23 @@ function filter_tags_generic(kv)
...
   if ((kv["hov"] and kv["hov"] ~= "no") or kv["hov:lanes"] or kv["hov:minimum"]) then
 
     kv["hov_tag"] = "true"
 
-    if (kv["hov"] == "designated") then
+    local only_hov_allowed = kv["hov"] == "designated"
+    if only_hov_allowed then
+      if kv["hov:lanes"] then
+        for lane in (kv["hov:lanes"] .. '|'):gmatch("([^|]*)|") do
+          if lane and lane ~= "designated" then
```
Comment (1ec5) [inline lua/graph.lua]: FYI, for the purpose of routing, `yes` and `designated` are more or less equivalent within access tags. So as far as I can tell, `yes` should be treated similarly to `designated`, and `designated` shouldn’t by itself prohibit any mode of transportation in particular.

/ref https://github.com/valhalla/valhalla/issues/1297#issuecomment-391767406

**Comment by chaupow** [comment]
@gknisely what's the status here? Can I hand this over to you - or can you ping me when you finished code review?

**Comment by gknisely** [comment]
@chaupow I ran the RAD tests and I was seeing some odd bicycle routes.  I don't think they are related to the HOV change; however, I want to confirm that first.  Yes I can take this over.  No problem.

**Comment by gknisely** [comment]
@chaupow looks great.  Bike diffs were from elevation data missing from my local data set.

**Pull request merged**